File: 백준/Silver/14469. 소가 길을 건너간 이유 3/소가 길을 건너간 이유 3.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        List<int[]> arr = new ArrayList<>();
        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int[] temp = new int[2];
            temp[0] = Integer.parseInt(st.nextToken());
            temp[1] = Integer.parseInt(st.nextToken());
            arr.add(temp);
        }
        arr.sort(Comparator.comparingInt((int[] a) -> a[0]).thenComparingInt(a -> a[1]));

        int time = -1;
        for (int i = 0; i < N; i++) {
            time = (arr.get(i)[0] >= time) ? arr.get(i)[0] + arr.get(i)[1] : time + arr.get(i)[1];
        }
        System.out.println(time);
    }
}

File: 백준/Silver/14469. 소가 길을 건너간 이유 3/소가 길을 건너간 이유 3.py
import sys
input = sys.stdin.readline

def solution():
    N = int(input().strip())
    arr = [list(map(int, input().strip().split())) for _ in range(N)]
    arr.sort(key=lambda x: (x[0], x[1]))

    time = -1
    for i in range(N):
        if arr[i][0] >= time:
            time = arr[i][0] + arr[i][1]
        else:
            time += arr[i][1]
    print(time)


solution()